pil-analyzer/src/statement_processor.rs

```diff
@@ -428,18 +445,56 @@ where
                     length: None,
                     degree: None,
                 };
-                let value = FunctionValueDefinition::TypeConstructor(
+                let value = FunctionValueDefinition::TypeConstructor(TypeConstructor::Enum(
                     shared_enum_decl.clone(),
                     variant.clone(),
-                );
+                ));
                 PILItem::Definition(var_symbol, Some(value))
             });
             return iter::once(PILItem::Definition(
                 symbol,
-                Some(FunctionValueDefinition::TypeDeclaration(enum_decl.clone())),
+                Some(FunctionValueDefinition::TypeDeclaration(
+                    TypeDeclarationAnalyzed::Enum(enum_decl.clone()),
+                )),
             ))
             .chain(var_items)
             .collect();
+        } else if let Some(FunctionDefinition::TypeDeclaration(TypeDeclarationParsed::Struct(
```

Comment:
I think at this point, we should really split out the structs and enums into functions. This function is getting too long. Also isn't there any code that can be shared?